### PR TITLE
INT-12 Added functionality to show job vacancy before submitting for,…

### PR DESCRIPTION
**Pull request by PH86**
… also allows to re-edit or clear the form

### CODE REVIEW

File: src/components/VacancyForm/VacancyForm.tsx [hunk 1]
```diff
@@ -258,7 +268,8 @@ export const VacancyForm: React.FC<{}> = (): React.ReactElement => {
                         </div>
                     </div>
                 </div>
-                <button type="submit" className='standard-btn'>Create Job</button>
+                <button type="submit" className='standard-btn'>Review Job</button>
```
Comment (JackHewitt08) [inline src/components/VacancyForm/VacancyForm.tsx]: no short names please btn --> button

File: src/components/VacancyFull/VacancyFull.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            salaryMax: '',
+            endDate: '',
+            jobDescription: '',
+            essentialRequirments: [],
+            desiredRequirments: [],
+            responsibilities: []
+        });
+        setShowVacancy(false);
+        setOpenModal(false);
+    }
```
Comment (JackHewitt08) [inline src/components/VacancyFull/VacancyFull.tsx:22]: line before return statement

File: src/components/VacancyFull/VacancyFull.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { AppContext } from "../../context";
+
+export const VacancyFull: React.FC<{}> = (): React.ReactElement => {
+    const { setOpenModal, setShowVacancy, jobForm, setJobForm } = React.useContext(AppContext);
+
```
Comment (JackHewitt08) [inline src/components/VacancyFull/VacancyFull.tsx]: remove extra line

File: src/components/VacancyFull/VacancyFull.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                    <ul>
+                        {jobForm.responsibilities.map(responsibility => {
+                            return <li>{responsibility}</li>
+                        })}
+                    </ul>
+                </div>
+                <button onClick={() => submitForm()} className="standard-btn">Submit</button>
+                <button onClick={() => setShowVacancy(false)} className="standard-btn">Edit</button>
+            </article>
+
```
Comment (JackHewitt08) [inline src/components/VacancyFull/VacancyFull.tsx]: remove extra line

File: src/App.tsx [hunk 5]
```diff
@@ -15,9 +15,23 @@ import { SingleJob } from "./pages/SingleJob/SingleJob";
...
+	const [jobForm, setJobForm] = React.useState({
+		jobTitle: '',
+		companyName: '',
+		location: '',
+		companyDescription: '',
+		salaryMin: '',
+		salaryMax: '',
+		endDate: '',
+		jobDescription: '',
+		essentialRequirments: [''],
```
Comment (JackHewitt08) [inline src/App.tsx]: i think default state here can just be an empty array, I think this will give it an extra element of empty string

**Review by JackHewitt08 — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 6]
```diff
@@ -15,9 +15,23 @@ import { SingleJob } from "./pages/SingleJob/SingleJob";
 const App: React.FC<{}> = (): React.ReactElement => {
 	const [loggedIn, setLoggedIn] = React.useState(false);
```
Comment (sambuckley1995) [inline src/App.tsx:16]: could we not just import { useState } from "react" to avoid having to declare React.something across multiple lines?

Comment (JackHewitt08) [reply src/App.tsx:16]: It's mainly there to be explicit, that's my preference if I'm honest

Comment (sambuckley1995) [reply src/App.tsx:16]: fair, I can't argue against that

File: src/App.tsx (hunk 5 above)
Comment (PH86) [reply src/App.tsx]: If I remove the empty string from the array I get  the error 

Type 'Dispatch<SetStateAction<{ jobTitle: string; companyName: string; location: string; companyDescription: string; salaryMin: string; salaryMax: string; endDate: string; jobDescription: string; essentialRequirments: never[]; desiredRequirments: never[]; responsibilities: never[]; }>>' is not assignable to type '(jobForm: { jobTitle: string; companyName: string; location: string; companyDescription: string; salaryMin: string; salaryMax: string; endDate: string; jobDescription: string; essentialRequirments: string[]; desiredRequirments: string[]; responsibilities: string[]; }) => void'.
  Types of parameters 'value' and 'jobForm' are incompatible.
    Type '{ jobTitle: string; companyName: string; location: string; companyDescription: string; salaryMin: string; salaryMax: string; endDate: string; jobDescription: string; essentialRequirments: string[]; desiredRequirments: string[]; responsibilities: string[]; }' is not assignable to type 'SetStateAction<{ jobTitle: string; companyName: string; location: string; companyDescription: string; salaryMin: string; salaryMax: string; endDate: string; jobDescription: string; essentialRequirments: never[]; desiredRequirments: never[]; responsibilities: never[]; }>'.
      Type '{ jobTitle: string; companyName: string; location: string; companyDescription: string; salaryMin: string; salaryMax: string; endDate: string; jobDescription: string; essentialRequirments: string[]; desiredRequirments: string[]; responsibilities: string[]; }' is not assignable to type '{ jobTitle: string; companyName: string; location: string; companyDescription: string; salaryMin: string; salaryMax: string; endDate: string; jobDescription: string; essentialRequirments: never[]; desiredRequirments: never[]; responsibilities: never[]; }'.
        Types of property 'essentialRequirments' are incompatible.
          Type 'string[]' is not assignable to type 'never[]'.
            Type 'string' is not assignable to type 'never'.

File: src/components/VacancyFull/VacancyFull.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+            desiredRequirments: [],
+            responsibilities: []
+        });
+        setShowVacancy(false);
+        setOpenModal(false);
+    }
+    return (
+        <div>
+            <h1>Vacancy render test</h1>
+            <article className='single-job-container' >
```
Comment (sambuckley1995) [inline src/components/VacancyFull/VacancyFull.tsx]: whitespace after the className

**Review by JackHewitt08 — APPROVED** [review]

**Review by JackHewitt08 — APPROVED** [review]

**Review by sambuckley1995 — APPROVED** [review]

**Review by sambuckley1995 — APPROVED** [review]

**Pull request merged**